sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/pipeOperators.scala

```diff
@@ -57,6 +66,21 @@ case class PipeExpression(child: Expression, isAggregate: Boolean, clause: Strin
   }
 }
 
+/**
+ * Validates and strips PipeExpression nodes from a logical plan once the child expressions are
+ * resolved.
+ */
+object ValidateAndStripPipeExpressions extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUpWithPruning(
+    _.containsPattern(PIPE_EXPRESSION), ruleId) {
+    case node: LogicalPlan =>
+      node.resolveExpressions {
+        case p: PipeExpression if p.child.resolved =>
+          p.checkInvariantsAndRemove
```

Comment:
shall we remove this lazy val and inline the validation code here?

Reply:
good idea, done.